### ISSUE TITLE
Feature request: Add support for empty buffers to lws_write

### ISSUE THREAD

**Issue by emptyVoid** [warmcat/libwebsockets#477]
ATM to close connection I have to create some dummy buffer for LWS not to crush.
Would be great to be able to do something like this:
```
lws_write( wsi, nullptr, 0, LWS_WRITE_CLOSE );
```

**Comment by lws-team** [comment]
You should only close things by returning -1 from the callback... you can control the close code issue like this (it already handles NULL payload)

```
			lws_close_reason(wsi, LWS_CLOSE_STATUS_GOINGAWAY, NULL, 0);
```

**Comment by emptyVoid** [comment]
Sorry, there's no such function on version 1.6.

**Comment by lws-team** [comment]
What you want exists in 1.7.

There's no way to add lws_close_reason() to v1.6 because it's a new api / api change.

So your choices are update to v1.7.x or try something like this on top

```
diff --git a/lib/output.c b/lib/output.c
index b527e9c..1d68c89 100644
--- a/lib/output.c
+++ b/lib/output.c
@@ -233,6 +233,9 @@ handle_truncated_send:
  *     positive number reflecting the amount of bytes actually sent.  This
  *     can be less than the requested number of bytes due to OS memory
  *     pressure at any given time.
+ *
+ *     If you have nothing to say when closing, you may give a NULL @buf and
+ *     0 @len, a temp buffer will be used.
  */
 
 LWS_VISIBLE int lws_write(struct lws *wsi, unsigned char *buf, size_t len,
@@ -242,10 +245,16 @@ LWS_VISIBLE int lws_write(struct lws *wsi, unsigned char *buf, size_t len,
        int masked7 = (wsi->mode == LWSCM_WS_CLIENT);
        unsigned char is_masked_bit = 0;
        unsigned char *dropmask = NULL;
+       unsigned char dummy[LWS_PRE + LWS_PRE];
        struct lws_tokens eff_buf;
        int pre = 0, n;
        size_t orig_len = len;
 
+       assert(!(!buf && len));
+
+       if (!buf)
+               buf = &dummy[LWS_PRE];
+
        if (wsi->state == LWSS_ESTABLISHED && wsi->u.ws.tx_draining_ext) {
                /* remove us from the list */
                struct lws **w = &pt->tx_draining_ext_list;
```

**Comment by emptyVoid** [comment]
Thanks!
I guess I'm going to stick with my dummy buffer (anyway it's implemented in the same manner) and  upgrade to 1.7 at some point.

**Comment by lws-team** [comment]
OK good luck.
